server/src/immich/api-v1/asset/asset.service.ts

```diff
@@ -386,13 +410,16 @@ export class AssetService {
         await this.jobRepository.queue({ name: JobName.SEARCH_REMOVE_ASSET, data: { ids: [id] } });
 
         result.push({ id, status: DeleteAssetStatusEnum.SUCCESS });
-        deleteQueue.push(
-          asset.originalPath,
-          asset.webpPath,
-          asset.resizePath,
-          asset.encodedVideoPath,
-          asset.sidecarPath,
-        );
+
+        if (!asset.isReadOnly) {
+          deleteQueue.push(
+            asset.originalPath,
+            asset.webpPath,
+            asset.resizePath,
+            asset.encodedVideoPath,
+            asset.sidecarPath,
```

Comment:
When an asset is deleted through the UI we need to skip read only ones there as well.

Reply:
I didn't see any other calls to queue a delete in the code. The API endpoint for `/delete` just appears to call deleteAll. Could you point me in the direction of where the others may be?

Reply:
Also, I initially opted to put this into the job itself, but the only data currently in the queue are actual filepaths, there is no asset information passed into this job.

Reply:
I think we have some work to do to refactor how assets are deleted. Ideally using a queue by asset id.

Reply:
Oh you're right it does always call this delete all endpoint

Reply:
Marking this as resolved unless you want the refactor to happen in this PR.